src/analysis/multiSpecies/Microbiome_Modeling_Toolbox/PairwiseInteractionModeling/joinModelsPairwiseFromList.m

```diff
@@ -0,0 +1,127 @@
+function [pairedModels,pairedModelInfo] = joinModelsPairwiseFromList(modelList,inputModels,c,u,mergeGenes,numWorkers)
+% Joins a list of COBRA model structures in all possible combinations.
+%
+% USAGE:
+% [pairedModels,pairedModelInfo]=joinModelsPairwiseFromList(modelList,inputModels,c,u,mergeGenes,numWorkers)
+% This function joins a list of microbial genome-scale reconstructions in 
+% all combinations. Models are not paired with themselves and pairs are 
+% only created once (model1+model2 = model2+model1). The reactions in each
+% joined reconstruction structure receive a suffix (entered as input 
+% parameter model list).
+% Coupling constraints are created in which the flux through all reactions 
+% of each individual microbe is coupled to the flux through its own biomass
+% objective function. By default, coupling constraints are implemented with
+% a coupling factor c of 400 and a threshold u of 0. The coupling 
+% parameters can be changed by entering the input parameters c and u.
+% By default, the gene-protein-reaction associations of the joined models
+% are not merged as the gene associations are not needed for the simulation
+% of pairwise interactions and significantly slow down pairwise model
+% creation.  If merging of genes is desired, please set the input
+% parameter mergeGenes to true.
+%
+% INPUTS
+% modelList           Cell array with names of reconstruction structures to be 
+%                     joined
+% inputModels         Array with COBRA model structures to be joined (needs to
+%                     have same length as modelList)
+% OPTIONAL INPUTS
+% c                   Coupling factor by which reactions in each joined model are 
+%                     coupled to its biomass reaction (default: 400)  
+% u                   Threshold representing minimal flux allowed when flux 
+%                     through biomass reaction in zero (default: 0)
+% mergeGenes          If true, genes in the joined models are merged and included
+%                     in the joined model structure (default: false)
+% numWorkers          Number of workers in parallel pool if desired
+% OUTPUTS
+% pairedModels        Structue containing created pairwise models in all
+%                     combinations
+% pairedModelInfo     Table with information on created pairwise models
+
+% Author: 
+%   - Almut Heinken, 02/2018
+
+if nargin <3
```

Comment:
There are a number of optional parameters here. Personally I would suggest varargin and an inputParser. 
There are two advantages:
1. Future modification of this function will allow additional parameters being added without the problem of having to stick to a specific order for the input parameters (i.e. better backward compatability)
2. Changes which modify the inputs do not have to consider the position of the specific parameter, and if a later optional argument is should be set earlier arguments do not have to be considered.

At least change this from `nargin < X` to ~exist('parameterName','var'), which makes the code better readable.
@rmtfleming whats your opinion?